newrelic/hooks/framework_falcon.py

```diff
@@ -47,11 +47,11 @@ def wrap_handle_exception(wrapped, instance, args, kwargs):
             try:
                 resp = bind_handle_exception(*args, **kwargs)
                 response_code = int(resp.status.split()[0])
-                if ignore_status_code(response_code):
+                if should_ignore_error(exc_info, response_code):
                     return result
-                record_exception(*exc_info)
+                notice_error(exc_info)
```

Comment:
Similar to the Django instrumentation. The if statement should be removed and the response_code should be passed into notice_error, let it handle deciding to ignore or record the exception.